@xen-orchestra/fs/src/abstract.js

```diff
@@ -145,6 +146,38 @@ export default class RemoteHandlerAbstract {
     throw new Error('Not implemented')
   }
 
+  async rmdir(
+    dir: string,
+    { recursive = false }: { recursive?: boolean } = {}
+  ) {
+    await (recursive ? this._rmtree(dir) : this._rmdir(dir))
+  }
+
+  async _rmdir(dir: string) {
+    throw new Error('Not implemented')
+  }
+
+  async _rmtree(dir: string) {
+    try {
+      return await this._rmdir(dir)
+    } catch (error) {
+      if (error.code !== 'ENOTEMPTY') {
+        throw error
+      }
+    }
+
+    const files = await this._list(dir)
+    await asyncMap(files, file =>
+      this._unlink(`${dir}/${file}`).catch(error => {
+        if (error.code === 'EISDIR') {
+          return this._rmtree(`${dir}/${file}`)
+        }
+        throw error
+      })
+    )
+    this._rmtree(dir)
```

Comment:
Don't forget to return this promise or the recursion is broken.

Reply:
fix